exporters/dsexporter/dsexporter.go

```diff
@@ -1,20 +1,46 @@
 package main
 
 import (
-	"fmt"
 	"net/http"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promhttp"
 )
 
-func ping(w http.ResponseWriter, req *http.Request) {
-	fmt.Fprintf(w, "pong")
+// CustomExporter is a dummy exporter that implements the prometheus.Collector interface.
+// The Custom exporter is a way to structure our code so that it can generate and expose metrics.
+type CustomExporter struct {
+	pingCounter prometheus.Counter
+}
+
+// Creating new instance of CustomExporter.
+func NewCustomExporter() *CustomExporter {
+	return &CustomExporter{
+		pingCounter: prometheus.NewCounter(prometheus.CounterOpts{
+			Name: "ping_request_count",
+			Help: "Number of requests handled by Ping handler",
+		}),
+	}
+}
+
+// Describe method sends descriptions of the metrics to Prometheus.
+// When Prometheus scrapes the /metrics endpoint of the exporter,
+// it first calls the Describe method to get a description of all the metrics.
+func (e *CustomExporter) Describe(ch chan<- *prometheus.Desc) {
+	e.pingCounter.Describe(ch)
+}
+
+// Collect method sends the current values of the metrics to Prometheus.
+// After Prometheus understands what metrics are available (using the `Describe` method),
+// it then calls the `Collect` method to actually get the values of those metrics.
+func (e *CustomExporter) Collect(ch chan<- prometheus.Metric) {
+	e.pingCounter.Collect(ch)
 }
 
 func main() {
-	http.HandleFunc("/ping", ping)
+	exporter := NewCustomExporter()
+	prometheus.MustRegister(exporter)
```

Comment:
```suggestion
        reg := prometheus.NewPedanticRegistry()
	exporter := NewCustomExporter()
	reg.MustRegister(exporter)

	// http.Handle("/metrics", promhttp.Handler())
	// http.ListenAndServe(":8090", nil)
	http.Handle("/metrics", promhttp.HandlerFor(
		reg,
		promhttp.HandlerOpts{
			EnableOpenMetrics: true,
			Registry:          reg,
		},
	))
	http.ListenAndServe(":8090", nil)
```

When we are using the default registry, we get a lot of predefined metrics that we do not need (take a look at the /metrics endpoint).
We can instead use [NewPedanticRegistry](https://pkg.go.dev/github.com/prometheus/client_golang/prometheus#NewPedanticRegistry) to get a clean /metrics with only our metric

Reply:
Yes, I will take a look at the [NewPedanticRegistry](https://pkg.go.dev/github.com/prometheus/client_golang/prometheus#NewPedanticRegistry) to get a clean /metrics with only our metric